proxy/src/connection.rs

```diff
@@ -165,10 +164,15 @@ impl io::Write for Connection {
 
 impl AsyncWrite for Connection {
     fn shutdown(&mut self) -> Poll<(), io::Error> {
+        use std::net::Shutdown;
         use self::Connection::*;
 
         match *self {
-            Plain(ref mut t) => t.shutdown(),
+            Plain(ref mut t) => {
+                try_ready!(AsyncWrite::shutdown(t));
+                // tcp shutdown afters
```

Comment:
s/afters/after/?

Please expand this comment with an explanation of *why* we're doing this.

In general, it would be good to use complete sentences in comments, so that it is clear(er) if/when a comment is complete.